src/OpenTelemetry.Extensions.PersistentStorage.Abstractions/PersistentStorageAbstractionsEventSource.cs

```diff
@@ -0,0 +1,108 @@
+﻿// <copyright file="PersistentStorageAbstractionsEventSource.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Diagnostics.Tracing;
+using System.Runtime.CompilerServices;
+
+namespace OpenTelemetry.Extensions.PersistentStorage
+{
+    [EventSource(Name = EventSourceName)]
+    internal sealed class PersistentStorageAbstractionsEventSource : EventSource
+    {
+        public static PersistentStorageAbstractionsEventSource Log = new PersistentStorageAbstractionsEventSource();
+        private const string EventSourceName = "OpenTelemetry-Extensions-PersistentStorage-Abstractions";
+
+        [NonEvent]
```

Comment:
why do we need this nonevent which calls the actual event?

Reply:
https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/OpenTelemetryProtocolExporterEventSource.cs follow this example. use noneevents for avoiding allocation cost for exception scenarios. for everything else, use the event direclty

Reply:
@cijothomas - Please review if it looks good now.